src/worker/keyman-api-service.ts

```diff
@@ -0,0 +1,36 @@
+import { keyboardDataObj, KeymanObj } from "./models";
+
+export class KeymanApi {
+  /**
+   * API configs
+   */
+  baseUrl = "https://api.keyman.com/search";
+  /**
+   * Query data to searche only for languages that match
+   */
+  params = { q: "l" };
+  url: URL;
+  langageArray: Array<keyboardDataObj>;
+
+  constructor() {
+    this.langageArray = [];
```

Comment:
Why is this a mutable property of this class? Why does it get written to (via `.push()`) and returned in the same method, but is never emptied or accessed otherwise by the class?

Reply:
I'm still waiting for an explaination for this one. Is it intended that when you call `fetchLanguageData()` more than once, that the languageArray may potentially have many duplicate languages?